spectec/src/exe-watsup/main.ml

```diff
@@ -65,12 +98,8 @@ let argspec = Arg.align
   "--print-il", Arg.Set print_elab_il, " Print il (after elaboration)";
   "--print-final-il", Arg.Set print_final_il, " Print final il";
   "--print-all-il", Arg.Set print_all_il, " Print il after each step";
-
-  "--sub", Arg.Set pass_sub, " Synthesize explicit subtype coercions";
-  "--totalize", Arg.Set pass_totalize, " Run function totalization";
-  "--the-elimination", Arg.Set pass_unthe, " Eliminate the ! operator in relations";
-  "--sideconditions", Arg.Set pass_sideconditions, " Infer side conditions";
-  "--animate", Arg.Set pass_animate, " Animate equality conditions";
+] @ List.map pass_argspec all_passes @ [
+  "--all-passes", Arg.Set run_all_passes, " Run all passes";
```

Comment:
What about appending every pass to `selected_passes` here, instead of using `run_all_pass` flag?
By doing so, fold_lefting over `all_passes` and do membership check inside, will simply be fold_lefting over `selected_passes`.

Reply:
I used a Set for `selected_passes` and removed the `run_all_passes : ref bool`; should be cleaner now.

Reply:
I see,
then it makes sense **not** to iterate over `seleceted_passes`.